flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/wmassigners/WatermarkAssignerOperator.java

```diff
@@ -52,12 +53,22 @@ public class WatermarkAssignerOperator extends AbstractStreamOperator<RowData>
 
     private transient long watermarkInterval;
 
+    private transient long timerInterval;
+
     private transient long currentWatermark;
 
-    private transient long lastRecordTime;
+    // Last time watermark have been (periodically) emitted
+    private transient long lastWatermarkPeriodicEmitTime;
+
+    // Last time idleness status has been checked
+    private transient long idleSince;
```

Comment:
The comment is fine, but this name is a bit ambiguous to me.

From the literal meaning, `idleSince` means when this operator is marked to IDLE. But it is the approximate time when the latest data was processed.

Reply:
timeSinceLastIdleCheck sounds make sense to me.